File: BuilderDesignPattern/CarManufacture_Example/Builders/CarBuilder.java
package BuilderDesignPattern.CarManufacture_Example.Builders;

import BuilderDesignPattern.CarManufacture_Example.Components.Engine;
import BuilderDesignPattern.CarManufacture_Example.Components.GPSNavigator;
import BuilderDesignPattern.CarManufacture_Example.Components.Transmission;
import BuilderDesignPattern.CarManufacture_Example.Components.TripComputer;
import BuilderDesignPattern.CarManufacture_Example.ProductTypes.CarsType;
import BuilderDesignPattern.CarManufacture_Example.Products.Car;

public class CarBuilder implements Builder<Car> {
	private String model;
	private CarsType type;
	private int seats;
	private Engine engine;
	private Transmission transmission;
	private TripComputer tripComputer;
	private GPSNavigator gpsNavigator;

	/**
	 * @param carModel
	 * string, name of the car
	 */
	@Override
	public void setCarModel(String carModel) {
		this.model = carModel;
	}

	/**
	 * @param carType
	 * enum for the type of car.
	 */
	public void setCarType(CarsType carType) {
		this.type = carType;
	}

	/**
	 * @param seats
	 * int for the seats of car.
	 */
	@Override
	public void setSeats(int seats) {
		this.seats = seats;
	}

	/**
	 * @param engine
	 * engine for the engine of car.
	 */
	@Override
	public void setEngine(Engine engine) {
		this.engine = engine;
	}

	/**
	 * @param transmission
	 * transmission for the transmission of car.
	 */
	@Override
	public void setTransmission(Transmission transmission) {
		this.transmission = transmission;
	}

	/**
	 * @param tripComputer
	 * tripComputer for the tripComputer of car.
	 */
	@Override
	public void setTripComputer(TripComputer tripComputer) {
		this.tripComputer = tripComputer;
	}

	/**
	 * @param gpsNavigator
	 * gpsNavigator for the gpsNavigator of car.
	 */
	@Override
	public void setGPSNavigator(GPSNavigator gpsNavigator) {
		this.gpsNavigator = gpsNavigator;
	}

	@Override
	public Car getResults() {
		return new Car(model, type, seats, engine, transmission, tripComputer, gpsNavigator);
	}
}

File: BuilderDesignPattern/CarManufacture_Example_Python/Builders/CarBuilder.py
from BuilderDesignPattern.CarManufacture_Example_Python.Builders.Builder import Builder
from BuilderDesignPattern.CarManufacture_Example_Python.Components.Engine import Engine
from BuilderDesignPattern.CarManufacture_Example_Python.Components.GPSNavigator import GPSNavigator
from BuilderDesignPattern.CarManufacture_Example_Python.Components.Transmission import Transmission
from BuilderDesignPattern.CarManufacture_Example_Python.Components.TripComputer import TripComputer
from BuilderDesignPattern.CarManufacture_Example_Python.ProductsTypes.CarsType import CarsType
from BuilderDesignPattern.CarManufacture_Example_Python.Products.Car import Car


class CarBuilder(Builder[Car]):
	_model: str
	_type: CarsType
	_seat: int
	_engine: Engine
	_transmission: Transmission
	_tripComputer: TripComputer
	_gpsNavigator: GPSNavigator

	def setCarModel(self, carName: str) -> Builder:
		self._model = carName
		return self

	def setCarType(self, carType: CarsType) -> Builder:
		self._type = carType
		return self

	def buildSeats(self, numSeats: int) -> Builder:
		self._seat = numSeats
		return self

	def buildEngine(self, carEngine: Engine) -> Builder:
		self._engine = carEngine
		return self

	def buildTransmission(self, carTransmission: Transmission) -> Builder:
		self._transmission = carTransmission
		return self
	
	def buildTripComputer(self, carTripComputer: TripComputer) -> Builder:
		self._tripComputer = carTripComputer
		return self

	def buildGPSNavigator(self, carGPSNav: GPSNavigator) -> Builder:
		self._gpsNavigator = carGPSNav
		return self

	def build(self) -> Car:
		return Car(self._model, self._type, self._seat, self._engine, self._transmission, self._tripComputer, self._gpsNavigator)